lib/matplotlib/widgets.py

```diff
@@ -2128,29 +2167,31 @@ def __init__(self, ax, onselect, direction, minspan=0, useblit=False,
 
         self.grab_range = grab_range
         self._interactive = interactive
+        self._edge_handles = None
         self.drag_from_anywhere = drag_from_anywhere
         self.ignore_event_outside = ignore_event_outside
 
         # Reset canvas so that `new_axes` connects events.
         self.canvas = None
-        self.artists = []
         self.new_axes(ax)
 
         # Setup handles
-        handle_props = {
+        self._handle_props = {
             'color': props.get('facecolor', 'r'),
             **cbook.normalize_kwargs(handle_props, Line2D._alias_map)}
```

Comment:
Can pass the type directly, I believe
```suggestion
            **cbook.normalize_kwargs(handle_props, Line2D)}
```